code/modules/mob/floating_messages.dm

```diff
@@ -0,0 +1,93 @@
+// Thanks to Burger from Burgerstation for the foundation for this
+
+/atom/movable
+	var/list/stored_chat_text
+
+/atom/movable/proc/animate_chat(message, datum/language/language, small, list/show_to, duration)
+	set waitfor = FALSE
+
+	var/style	//additional style params for the message
+	var/fontsize = 6
+	if(small)
+		fontsize = 5
+	if(copytext(message, length(message) - 1) == "!!")
+		fontsize = 8
+		style += "font-weight: bold;"
+
+	if(length(message) > 50)
+		message = "[copytext(message, 1, 50)]..."
+
+	style += "color: [get_floating_message_color(name)];"
```

Comment:
wouldn't this be kind of expensive to run every time someone says something

Reply:
no idea, someone smarter than me will probably think of something better, but i doubt this is too intense

Reply:
we have a whole 32 bits to deal with it, lmfao